model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapKeycloakTransaction.java

```diff
@@ -98,7 +103,8 @@ public E read(String key) {
         if (key == null) return null;
         UUID uuid = StringKeyConverter.UUIDKey.INSTANCE.fromStringSafe(key);
         if (uuid == null) return null;
-        return mapToEntityDelegateUnique(em.find(entityType, uuid));
+        E e = mapToEntityDelegateUnique(em.find(entityType, uuid));
+        return e != null && isExpirableEntity && isExpired((ExpirableEntity) e, true) ? null : e;
```

Comment:
If I got it right we should never obtain an entity which is actually expired, right? So this check is here to be sure, is that so?

Reply:
Yes, you are right. Ideally, Keycloak should not know about the presence of expired entities in the store.